### ISSUE TITLE
Messaging on project doesn't work correctly

### ISSUE THREAD

**Issue by vic-topcoder** [appirio-tech/connect-app#1691]
## Testing messaging and feed on the dev server, there are several issues:

1. Messages are not sorted in reverse chronological order based on user activity. The message that has a response to latest, should be on top. We count not the date of the topic created (unless there are no other events), but the last activity on a topic: created, responded, edited.

2. New posts do not trigger notifications or page reload recommendation.
Log in with 2 accounts, Pat Monahan and Christina Underwood, and open the same project ID on 2 browsers at the same time. Posting on one should trigger notification and page reload suggestion to the other.

3. Post authors are not correct. Posting new comments as Christina Underwood is displayed as Coder Bot on the other interface.

![screen shot 2018-02-14 at 3 34 05 pm](https://user-images.githubusercontent.com/8484061/36233805-9964b014-119c-11e8-9bcf-347f2a772393.png)

4. Timestamps are not correct. Posting from San Francisco causes the records to go 8 minutes in the past.

5. Bonus: Only latest project activity message should be displayed, we still have a lot of the project activity. Can we remove those from the system (filter them somehow?).

We have some strange limit on Discourse in Dev so I can't test topic creation. Can we increase the limit somehow?

**Comment by gondzo** [comment]
1. Only posting a topic/comment will affect sorting. editing a post will not. do you want both to count as topic updates and affect sorting?
2. I am getting notifications correctly in that same project. can you test again?
3. good catch, will fix
4. I can reproduce it, will look to fix it
5. That's part of #1538

**Comment by vic-topcoder** [comment]
> 1. Only posting a topic/comment will affect sorting. editing a post will not. do you want both to count as topic updates and affect sorting?

Yes, editing should affect timestamp. The logic is that any data added might be of value for the users. The timestamp should change to `edited X time ago`

I'm retesting, but still not getting page reload notification. Notifications seem to have lost the author (user)

**Comment by gondzo** [comment]
About the author, that was just for project changes (link, member changes, specification updates). fixed now

**Comment by vic-topcoder** [comment]
Two users looking at the same project (one is normal Safari, other in private browsing Firefox) see different feed.
![screen shot 2018-02-14 at 4 40 30 pm](https://user-images.githubusercontent.com/8484061/36235551-2c8e5d46-11a6-11e8-9214-e5b58e9d1de5.png)
